Cache char_equip data prior to processing in charutils::LoadEquip()

claywar opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
`charutils::LoadEquip()` was querying char_equip table, and then other functions could have sub-queries prior to exiting, which could clobber data.  This PR caches the results of the initial query prior to any other processing to prevent that.

<!-- Describe what your PR does here. If it closes an existing issue, you can mention: "Closes #1234" and GitHub will link this PR to that issue. -->

## Steps to test these changes
1. Equip an item in mainhand with a magian trial inscribed, and have other gear equipped
2. Log out
3. Log in

All equipment should be loaded successfully

![image](https://github.com/LandSandBoat/server/assets/81713309/9479ed8a-5c52-404e-bbfa-1ac90a7e8842)

Also confirmed that trials are unaffected, and proc as necessary.
<!-- Clear and detailed steps to test your changes here -->

Xaver-DaRed reviewed (approved):
This would fix weird stuff happening like characters getting fully naked on zone when zoning with a weapon with a trial?

claywar commented:
> This would fix weird stuff happening like characters getting fully naked on zone when zoning with a weapon with a trial?

Yes, if I can figure out why the win32 build hates me

WinterSolstice8 reviewed (approved):
LGTM but I'd add a comment on why we're caching because it's not something we'd normally do

claywar committed:
Cache char_equip data prior to processing

Remove debug comments, restore test code

Cast initial values

Add comment explaining reasoning for caching sql result
